### PR TITLE
drivers: ethernet: enc424j600: change mac addr runtime

### PR DESCRIPTION
**Pull request by jancoow**
Add ability to change mac address at runtime using net mgmt config call

### CODE REVIEW

File: drivers/ethernet/eth_enc424j600.c [hunk 1]
```diff
@@ -556,6 +556,41 @@ static enum ethernet_hw_caps enc424j600_get_capabilities(const struct device *de
...
+			tmp = config->mac_address.addr[2] | config->mac_address.addr[3] << 8;
+			enc424j600_write_sfru(dev, ENC424J600_SFR3_MAADR2L, tmp);
+
+			/* write MAC address byte 6 and 5 */
+			tmp = config->mac_address.addr[4] | config->mac_address.addr[5] << 8;
+			enc424j600_write_sfru(dev, ENC424J600_SFR3_MAADR3L, tmp);
+
+			return 0;
+		default:
+			return ENOTSUP;
```
Comment (pdgendt) [inline drivers/ethernet/eth_enc424j600.c:589]:
```suggestion
			return -ENOTSUP;
```

File: drivers/ethernet/eth_enc424j600.c [hunk 2]
```diff
@@ -556,6 +556,41 @@ static enum ethernet_hw_caps enc424j600_get_capabilities(const struct device *de
 	return ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T;
 }
 
+static int enc424j600_set_config(const struct device *dev,
+			    enum ethernet_config_type type,
+			    const struct ethernet_config *config)
```
Comment (pdgendt) [inline drivers/ethernet/eth_enc424j600.c:561]: Fix indentation

**Review by pdgendt — COMMENTED** [review]
Fix indentation please

File: drivers/ethernet/eth_enc424j600.c [hunk 3]
```diff
@@ -556,6 +556,41 @@ static enum ethernet_hw_caps enc424j600_get_capabilities(const struct device *de
 	return ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T;
 }
 
+static int enc424j600_set_config(const struct device *dev,
+                                 enum ethernet_config_type type,
+                                 const struct ethernet_config *config)
```
Comment (pdgendt) [inline drivers/ethernet/eth_enc424j600.c:561]: Please stick to the used indentation elsewhere (tabs + aligning spaces)

```suggestion
static int enc424j600_set_config(const struct device *dev,
				 enum ethernet_config_type type,
				 const struct ethernet_config *config)
```

Comment (jancoow) [reply drivers/ethernet/eth_enc424j600.c:561]: Well then i got an error that the line is to long.. I just copied everything from the get_config function so why do I need to fix the indent in the first place?

Comment (pdgendt) [reply drivers/ethernet/eth_enc424j600.c:561]: You had too many tabs (= 8 spaces) in the previous version.

Comment (jancoow) [reply drivers/ethernet/eth_enc424j600.c]: And apparently I cannot use the button commit suggestions.........................

**Review by pdgendt — APPROVED** [review]

**Comment by jancoow** [comment]
Sorry, I cannot get this compliance check to fix and I do not have time to keep fixing it.

**Comment by pdgendt** [comment]
> Sorry, I cannot get this compliance check to fix and I do not have time to keep fixing it. I give up.

Squash your commits into the first one (you shouldn't use fixup commits), and force push the single commit.

**Pull request closed without merging**